### PR TITLE
Allow to set predefined template parameters available in all templates

### PR DESCRIPTION
**Pull request by Hotas2k4**

### CODE REVIEW

File: test/EmailTemplatesTest/Options/TemplateServiceOptionsTest.php [hunk 1]
```diff
@@ -98,4 +98,14 @@ public function testSetGetEngine()
...
 
+    /**
+     * @covers ::setPredefinedParams
+     * @covers ::getPredefinedParams
+     */
+    public function testSetGetPredefinedParams()
+    {
+        $this->assertEquals([], $this->options->getPredefinedParams());
+        $this->options->setPredefinedParams(['url' => 'http://']);
+        $this->assertArrayHasKey('url', $this->options->getPredefinedParams());
```
Comment (Ocramius) [inline test/EmailTemplatesTest/Options/TemplateServiceOptionsTest.php]: Add an `assertSame` too

File: test/EmailTemplatesTest/Options/TemplateServiceOptionsTest.php [hunk 2]
```diff
@@ -98,4 +98,14 @@ public function testSetGetEngine()
         $this->assertEquals('B20', $this->options->getEngine());
     }
 
+    /**
+     * @covers ::setPredefinedParams
+     * @covers ::getPredefinedParams
+     */
+    public function testSetGetPredefinedParams()
```
Comment (Ocramius) [inline test/EmailTemplatesTest/Options/TemplateServiceOptionsTest.php]: An additional test for verifying precedence of merged values is needed

File: src/Roave/EmailTemplates/Options/TemplateServiceOptions.php [hunk 3]
```diff
@@ -121,4 +128,20 @@ public function getEngine()
     {
         return $this->engine;
     }
+
+    /**
+     * @param array $predefinedParams
+     */
+    public function setPredefinedParams(array $predefinedParams)
```
Comment (Ocramius) [inline src/Roave/EmailTemplates/Options/TemplateServiceOptions.php]: Getter/setter unit tests needed too. Yep.

File: src/Roave/EmailTemplates/Service/TemplateService.php [hunk 4]
```diff
@@ -123,7 +128,8 @@ public function __construct(
      */
     public function render($templateId, $locale, array $parameters = array())
     {
-        $template = $this->repository->getByIdAndLocale($templateId, $locale);
+        $template   = $this->repository->getByIdAndLocale($templateId, $locale);
+        $parameters = array_merge($this->options->getPredefinedParams(), $parameters);
```
Comment (Ocramius) [inline src/Roave/EmailTemplates/Service/TemplateService.php]: Please don't overwrite the variable - change its name so that the semantics are clear

**Comment by Ocramius** [comment]
@Hotas2k4 can you prepare release `1.2.0 or should I?

**Comment by macnibblet** [comment]
He can't but I can

**Pull request merged**